e2e/head_test.go

```diff
@@ -0,0 +1,407 @@
+
+func TestHeadObjectJSONNonExistent(t *testing.T) {
+	t.Parallel()
+
+	_, s5cmd := setup(t)
+
+	cmd := s5cmd("--json", "head", "s3://non-existent-bucket/non-existent-file.txt")
+	result := icmd.RunCmd(cmd)
+
+	result.Assert(t, icmd.Expected{ExitCode: 1})
+}
+
+// head object --humanize
+
+func TestHeadObjectHumanize(t *testing.T) {
+	t.Parallel()
+
+	s3client, s5cmd := setup(t)
+
+	bucket := s3BucketFromTestName(t)
+	createBucket(t, s3client, bucket)
+
+	largeFileContent := make([]byte, 1024*1024*50) // 50MB
+	for i := 0; i < len(largeFileContent); i++ {
+		largeFileContent[i] = 'a'
+	}
+	putFile(t, s3client, bucket, "file.txt", string(largeFileContent))
+
+	cmd := s5cmd("head", "--humanize", fmt.Sprintf("s3://%v/file.txt", bucket))
+	result := icmd.RunCmd(cmd)
+
+	datePattern := `\d{4}/\d{2}/\d{2} \d{2}:\d{2}:\d{2}`
+	storageClassPattern := `(?:STANDART|)`
+	sizePattern := `(?:\d+(\.\d+)?[KMGTP]?B?)`
+	s3urlPattern := `file.txt`
+
+	expectedOutput := fmt.Sprintf(
+		`%s\s+%s\s+%s\s+%s`,
+		datePattern,
+		storageClassPattern,
+		sizePattern,
+		s3urlPattern,
+	)
+
+	match, _ := regexp.MatchString(expectedOutput, result.Combined())
+
+	if !match {
+		t.Errorf("expected output to match:\n%s\nbut got:\n%s", expectedOutput, result.Combined())
+	}
+
+	result.Assert(t, icmd.Expected{ExitCode: 0})
+
+}
+
+// head object --etag
+
+func TestHeadObjectEtag(t *testing.T) {
+	t.Parallel()
+
+	s3client, s5cmd := setup(t)
+
+	bucket := s3BucketFromTestName(t)
+	createBucket(t, s3client, bucket)
+	putFile(t, s3client, bucket, "file.txt", "content")
+
+	cmd := s5cmd("head", "--etag", fmt.Sprintf("s3://%v/file.txt", bucket))
+	result := icmd.RunCmd(cmd)
+
+	//  2024/07/03 13:13:15 STANDART 9a0364b9e99bb480dd25e1f0284c8555                  7  file.txt
+
+	datePattern := `\d{4}/\d{2}/\d{2} \d{2}:\d{2}:\d{2}`
+	storageClassPattern := `(?:STANDART|)`
+	sizePattern := `\d+`
+	s3urlPattern := `file.txt`
+	etagPattern := `[a-f0-9]+`
+
+	expectedOutput := fmt.Sprintf(
+		`%s\s+%s\s+%s\s+%s\s+%s`,
+		datePattern,
+		storageClassPattern,
+		etagPattern,
+		sizePattern,
+		s3urlPattern,
+	)
+
+	match, _ := regexp.MatchString(expectedOutput, result.Combined())
+
+	if !match {
+		t.Errorf("expected output to match:\n%s\nbut got:\n%s", expectedOutput, result.Combined())
+	}
+
+	result.Assert(t, icmd.Expected{ExitCode: 0})
+}
+
+// head object --show-fullpath
+
+func TestHeadObjectShowFullpath(t *testing.T) {
+	t.Parallel()
+
+	s3client, s5cmd := setup(t)
+
+	bucket := s3BucketFromTestName(t)
+	createBucket(t, s3client, bucket)
+
+	putFile(t, s3client, bucket, "file.txt", "content")
+
+	cmd := s5cmd("head", "--show-fullpath", fmt.Sprintf("s3://%v/file.txt", bucket))
+	result := icmd.RunCmd(cmd)
+
+	s3urlPattern := fmt.Sprintf(`s3://%s/file.txt`, bucket)
+
+	expectedOutput := s3urlPattern
+
+	match, _ := regexp.MatchString(expectedOutput, result.Combined())
+
+	if !match {
+		t.Errorf("expected output to match:\n%s\nbut got:\n%s", expectedOutput, result.Combined())
+	}
```

Comment:
not specifically related to this but to give an advice;

never ignore errors in tests, you can always fail with `t.Fatal`. if the regex pattern is constant and expected to be valid either in code itself of tests, it is better to use `regexp.MustCompile` for such patterns. (IMHO)